src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +893,68 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode", (e,v)->{
+            if(!(e instanceof ServerPlayerEntity)){
+                return;
+            }
+
+            switch(v.getString().toLowerCase()){
+                case "survival":((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);break;
+                case "creative":((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);break;
+                case "adventure":((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);break;
+                case "spectator":((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);break;
+            }
+
+            switch((int)NumericValue.asNumber(v).getLong()){
+                case 0:((ServerPlayerEntity) e).setGameMode(GameMode.SURVIVAL);break;
+                case 1:((ServerPlayerEntity) e).setGameMode(GameMode.CREATIVE);break;
+                case 2:((ServerPlayerEntity) e).setGameMode(GameMode.ADVENTURE);break;
+                case 3:((ServerPlayerEntity) e).setGameMode(GameMode.SPECTATOR);break;
+            }
+        });
+
+        put("jumping",(e,v)->{
+            if(!(e instanceof LivingEntity)) return;
+
+            ((LivingEntity) e).setJumping(v.getBoolean());
+        });
+
+        put("jump",(e,v)->{//Most *REDACTED* up code in the world
```

Comment:
I agree - should have made the jump() call public and just call it here.

Reply:
nah, that would have made it too simple

Reply:
In fact, I added it to MobAi.java (best place I could think to put it in, also cos you may need it when (and if) you r gonna add pathfinding to scarpet) I added it to carpet.utils, not carpet.script.utils, cos I thought that it felt more appropriate in MobAi as opposed to WorldTools